CliFx/Domain/CommandArgumentSchema.cs

```diff
@@ -176,4 +159,71 @@ internal partial class CommandArgumentSchema
                 BindingFlags.Public | BindingFlags.Static,
                 null, new[] {typeof(string), typeof(IFormatProvider)}, null);
     }
+
+    // Default and valid value handling.
+    internal partial class CommandArgumentSchema
+    {
+        public IReadOnlyList<string> GetValidValues()
+        {
+            var result = new List<string>();
+
+            // Some arguments may have this as null due to a hack that enables built-in options
+            if (Property == null)
+                return result;
+
+            var underlyingPropertyType =
+                Property.PropertyType.GetNullableUnderlyingType() ?? Property.PropertyType;
+
+            // Enum
+            if (underlyingPropertyType.IsEnum)
+                result.AddRange(Enum.GetNames(underlyingPropertyType));
+
+            return result;
+        }
+
+        public string? GetDefaultValue(object? instance)
+        {
+            if (Property is null || instance is null)
+            {
+                return null;
+            }
+
+            var instanceType = instance.GetType();
+            var commandType = Property?.DeclaringType;
+
+            if (instanceType != commandType)
+            {
+                throw new ArgumentException($"Argument {nameof(instance)} of type {instanceType} " +
+                    $"must be the same as this command's type {nameof(commandType)}.");
+            }
+
+            var propertyName = Property?.Name;
+            string? defaultValue = null;
+            var culture = CultureInfo.CurrentCulture;
+
+            if (!string.IsNullOrWhiteSpace(propertyName))
+            {
+                var instanceProperty = instance.GetType().GetProperty(propertyName);
+                var value = instanceProperty.GetValue(instance);
+                if (value.OverridesToStringMethod())
+                {
+                    defaultValue = value.ToCulturedString(culture).WrapWithQuotesIfEmptyOrWhiteSpace();
+                }
+                else if (value is IEnumerable values)
+                {
+                    var list = new List<string>();
+                    foreach (var val in values)
+                    {
+                        if (val is object)
+                        {                 
+                            list.Add(val.ToCulturedString(culture).WrapWithQuotesIfEmptyOrWhiteSpace());
+                        }
+                    }
+                    defaultValue = string.Join(" ", list);
```

Comment:
Could be simplified as:

```csharp
defaultValue = string.Join(" ", values.Where(v => v != null).Select(v => v.ToCulturedString(culture).WrapWithQuotesIfEmptyOrWhiteSpace());
```

Reply:
Done, I just had to cast `values` from `IEnumerable` to `IEnumerable<object>` first in order to be able to use LINQ. I left a comment on it to help others :)